implementations/rust/ockam/ockam_command/src/run/parser.rs

```diff
@@ -0,0 +1,323 @@
+use crate::Result;
+use duct::Expression;
+use ockam_api::cli_state::CliState;
+use ockam_core::compat::collections::HashMap;
+use once_cell::sync::Lazy;
+use serde::Deserialize;
+use std::collections::{BTreeMap, VecDeque};
+use std::fmt::Debug;
+use std::path::Path;
+use tracing::debug;
+
+pub struct ConfigRunner {
+    commands_sorted: Vec<ParsedCommand>,
+    commands_index: BTreeMap<String, usize>,
+}
+
+#[derive(Clone)]
+pub struct ParsedCommand {
+    pub id: String,
+    pub depends_on: Option<String>,
+    pub cmd: Expression,
+}
+
+impl ConfigRunner {
+    fn new() -> Self {
+        Self {
+            commands_sorted: vec![],
+            commands_index: Default::default(),
+        }
+    }
+
+    pub fn go(cli_state: &CliState, path: &Path) -> Result<()> {
+        let mut cr = Self::new();
+        cr.parse(cli_state, path)?;
+        cr.run()?;
+        Ok(())
+    }
+
+    fn parse(&mut self, cli_state: &CliState, path: &Path) -> Result<()> {
+        let config = std::fs::read_to_string(path)?;
+        let config: Config = serde_yaml::from_str(&config)?;
+        let mut nodes = VecDeque::new();
+        for (name, node) in config.nodes {
+            nodes.push_back((name, node));
+        }
+        while let Some((name, node)) = nodes.pop_front() {
+            // If the node depends on another node, check if that node has been parsed.
+            if let Some(depends_on) = &node.depends_on {
+                // If the node hasn't been parsed, push it back to the queue.
+                if !self
```

Comment:
What about reciprocal dependencies?
I would at least panic or better return an error

Reply:
I think I have an idea to control this without complicating the logic too much... I'll try it out.

Reply:
nice!